ISM43362Interface.cpp

```diff
@@ -441,7 +441,7 @@ int ISM43362Interface::socket_recv(void *handle, void *data, unsigned size)
 
     if (!socket->connected) {
         _mutex.unlock();
-        return NSAPI_ERROR_CONNECTION_LOST;
+        return 0;
     }
```

Comment:
question: socket_recv is supposed to return
     *  @return         Number of received bytes on success, negative error
     *                  code on failure
here, if the socket is not valid, shouldn't it be considered a failure ?